docs/onclusterlayering-quickstart.md

```diff
@@ -0,0 +1,486 @@
+State: idle
+Deployments:
+* ostree-unverified-registry:registry.ci.openshift.org/ocp/4.17-2024-08-19-220527@sha256:f686a353e75b2cf0981d69ee07233e71874ae0a1f37c10d883be070e63e5b60f
+                    Digest: sha256:f686a353e75b2cf0981d69ee07233e71874ae0a1f37c10d883be070e63e5b60f
+                  Version: 417.94.202408190141-0 (2024-08-19T01:45:32Z)
+
+Removing debug pod ...
+```
+
+We can see that we have the "factory" OS image that was provided by the current OpenShift release. Now, lets add the worker node to our `layered` MachineConfigPool by adding the `node-role.kubernetes.io/layered=` label to it so that it can use our newly-built OS image.
+
+
+```console
+$ oc label node/ip-10-0-18-226.ec2.internal 'node-role.kubernetes.io/layered='
+```
+
+Now, we'll see the node drain and begin to rebase into the newly-built OS image. This will take a few minutes, but you can use this script that will wait for it to complete:
+
+```bash
+#!/usr/bin/env bash
+
+nodeName="ip-10-0-18-226.ec2.internal"
+
+oc wait \
+    --timeout=10m \
+    --for=jsonpath='{.metadata.annotations.machineconfiguration\.openshift\.io/desiredImage=}{.metadata.annotations.machineconfiguration\.openshift\.io/currentImage}' \
+    --for=jsonpath='{.metadata.annotations.machineconfiguration\.openshift\.io/state}=Done' \
+    "node/$nodeName"
+```
+
+Once the node has finished the update, we can interrogate the node state like we did before:
+
+```console
+$ oc debug node/ip-10-0-18-226.ec2.internal -- chroot /host rpm-ostree status
+
+Starting pod/ip-10-0-18-226ec2internal-debug-xvbx4 ...
+To use host binaries, run `chroot /host`
+
+
+State: idle
+Deployments:
+* ostree-unverified-registry:image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-images@sha256:c47856f56e1fdb7c9d10a1658e4ea85fbea44d71fb0e82898d152b47e0f894c6
+                    Digest: sha256:c47856f56e1fdb7c9d10a1658e4ea85fbea44d71fb0e82898d152b47e0f894c6
+                  Version: 417.94.202408190141-0 (2024-08-20T14:40:42Z)
+
+
+
+Removing debug pod ...
+```
+
+We can see now that our node has booted into our new OS image. But does it have `cowsay` installed?
+
+
+```console
+$ oc debug node/ip-10-0-18-226.ec2.internal -- chroot /host rpm-ostree status
+
+Starting pod/ip-10-0-18-226ec2internal-debug-6vvvf ...
+To use host binaries, run `chroot /host`
+
+
+  ______________________ 
+< Hello layered world! >
+  ---------------------- 
+        \   ^__^
+          \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
+
+
+
+Removing debug pod ...
+```
+
+## Conclusion
+
+At this point, we now have a customized OS image installed on our cluster nodes. If the MachineConfigs for the `layered` MachineConfigPool are changed or the `Containerfile` is changed, a new `MachineOSBuild` will be created, the build will automatically start, and the image will be rolled out automatically to all of the nodes within the `layered` MachineConfigPool.
```

Comment:
> the `Containerfile` is changed
Or more generically if the `MachineOSConfig` is changed?

Reply:
Great point!